### PR TITLE
Update tests to use assertj (#2147)

### PR DESCRIPTION
**Pull request by pavanbaloju**
# pull request for (#2147 )

Hi @sanderploegsma, I have updated the following tests in this PR. Please review!

1. CollatzCalculatorTest.java
2. CryptoSquareTest.java
3. DartsTest.java
4. DifferenceOfSquaresCalculatorTest.java
5. DiffieHellmanTest.java
6. DnDCharacterTest.java
7. DominoesTest.java
8. ProteinTranslatorTest.java
9. RotationalCipherTest.java

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

File: exercises/practice/diffie-hellman/src/test/java/DiffieHellmanTest.java
Comment (sanderploegsma) [inline exercises/practice/diffie-hellman/src/test/java/DiffieHellmanTest.java:1]: Could you revert the changes to this file? The `diffie-hellman` exercise is deprecated, which means that [the exercise files should remain untouched](https://exercism.org/docs/building/tracks/deprecated-exercises)

**Review by sanderploegsma — CHANGES_REQUESTED** [review]
One small remark to take care of, the rest of the changes look good to me!

**Review by sanderploegsma — APPROVED** [review]
Nice work, thanks for your contribution!

**Comment by pavanbaloju** [comment]
> Nice work, thanks for your contribution!

Thanks for the oppurtunity!
